### ISSUE TITLE
hand_verts_new for mode 'train'

### ISSUE THREAD

**Issue by taeyunwoo** [rongakowang/DenseMutualAttention#2]
Hello, I'm trying to revise your code for training.
Before questions, I'm really appreciate your work! Thanks a lot.

Anyway, when the mode is 'train' in Model, the hand_verts_new variable was not assigned before.
Does it work similar to mode='test'?
In other words, the 'hand_verts_new' variable is from the estimated from hand_x?

Thank you for reading this question.

**Comment by rongakowang** [comment]
Hi,

Please ignore the condition about the `hand_verts_new` in [here](https://github.com/rongakowang/DenseMutualAttention/blob/main/main/model_no_render.py#L154), as in training you also need it for the rendering loss.

**Comment by taeyunwoo** [comment]
However, during training procedure, does not the rendering loss need in the training?

**Comment by rongakowang** [comment]
Please refer to the Eq. (24) in the paper for the rendering loss, you will need to recover the hand vertices in the Euclidean space in order to render the image

**Comment by taeyunwoo** [comment]
Thank you for replying. Furthermore, [hand_verts_z](https://github.com/rongakowang/DenseMutualAttention/blob/main/main/model_no_render.py#L158) and [hand_joint_z](https://github.com/rongakowang/DenseMutualAttention/blob/main/main/model_no_render.py#L158) occur errors about the dimension mismatch.

As I think, `targets['root_joint_depth'][2].unsqueeze(1).repeat(1, 778 (or 21) , 1)` may solve this problem. Is it correct to change these code parts to add only z-axis of hand_root to hand pose?

**Comment by rongakowang** [comment]
The `root_joint_depth` for both DexYCB and HO3D dataset is already a scalar as from `hand_root[2]` (see `__getitem` in both file), could you confirm this as I haven't faced dimension mismatch issue?

**Comment by taeyunwoo** [comment]
Currently, I am working with HO-3D, thus I did not see the code of Dex-YCB's `__getitem__`.

Anyway, I looked around the code for `get_root` and ho3d `__getitem__`.
However, I think the `get_root` in ho3d does not return the root joint (index: 0 in ho3d), instead, it returns all 21 joints of the hand.

**Comment by rongakowang** [comment]
I see, In HO3D for the testing the `handJoints3D` ia only for the root, while for training it contains all 21 joints (see its README), the dataset is for the testing only thus you might need to modify `get_root` if you wish to do the training

**Comment by taeyunwoo** [comment]
Sorry for bothering you about this problem. Still, I tried this code with 'train' mode of HO3D dataset and its loader. 
Thank you for your replying!

**Comment by kyxxc** [comment]
Hello,

Are you planning to release the training code?

I want to use it for my research.

Thanks.

**Comment by AnshuDFKI** [comment]
> Sorry for bothering you about this problem. Still, I tried this code with 'train' mode of HO3D dataset and its loader. Thank you for your replying!

Hi @taeyunwoo, were you able to generate training code? I am trying the same. If it is alright may be we can connect and help each other. I am also facing some problems. Please let me know.
